simple_parsing/helpers/partial.py

```diff
@@ -0,0 +1,308 @@
+                f"Ignoring this argument."
+            )
+            continue
+
+        class_help_entries = {v for k, v in class_docstring_help.items() if k.startswith(name)}
+        init_help_entries = {v for k, v in class_constructor_help.items() if k.startswith(name)}
+        help_entries = init_help_entries or class_help_entries
+        if help_entries:
+            help_str = help_entries.pop()
+        else:
+            help_str = ""
+
+        if default is dataclasses.MISSING:
+            field = simple_parsing.field(help=help_str, required=True)
+            # insert since fields without defaults need to go first.
+            fields.insert(0, (name, field_type, field))
+            logger.debug(f"Adding required field: {fields[0]}")
+        else:
+            field = simple_parsing.field(default=default, help=help_str)
+            fields.append((name, field_type, field))
+            logger.debug(f"Adding optional field: {fields[-1]}")
+
+    cls_name = _get_generated_config_class_name(cls)
+    config_class = make_dataclass(cls_name=cls_name, bases=(Partial,), fields=fields)
+    config_class._target_ = cls
+    config_class.__doc__ = (
+        f"Auto-Generated configuration dataclass for {cls.__module__}.{cls.__qualname__}\n"
+        + (cls.__doc__ or "")
+    )
+
+    return config_class
+
+
+def _parse_args_from_docstring(docstring: str) -> dict[str, str]:
+    """Taken from `pytorch_lightning.utilities.argparse`."""
+    arg_block_indent = None
+    current_arg = ""
+    parsed = {}
+    for line in docstring.split("\n"):
+        stripped = line.lstrip()
+        if not stripped:
+            continue
+        line_indent = len(line) - len(stripped)
+        if stripped.startswith(("Args:", "Arguments:", "Parameters:")):
+            arg_block_indent = line_indent + 4
+        elif arg_block_indent is None:
+            continue
+        elif line_indent < arg_block_indent:
+            break
+        elif line_indent == arg_block_indent:
+            current_arg, arg_description = stripped.split(":", maxsplit=1)
+            parsed[current_arg] = arg_description.lstrip()
+        elif line_indent > arg_block_indent:
+            parsed[current_arg] += f" {stripped}"
+    return parsed
+
+
+try:
+    # This only seems to be necessary for the SGD optimizer.
+    from torch.optim.optimizer import _RequiredParameter
+
+    @adjust_default.register(_RequiredParameter)
+    def _(default: Any) -> Any:
+        return dataclasses.MISSING
+
+except ImportError:
+    pass
+
+
+def _get_generated_config_class_name(target: type | Callable) -> str:
+    if inspect.isclass(target):
+        return target.__name__ + "Config"
+    elif inspect.isfunction(target):
+        return target.__name__ + "_config"
+    raise NotImplementedError(target)
+
+
+class _Partial(_ProtocolMeta):
+    _target_: _C
+
+    def __getitem__(cls, target: type[_T] | Callable[_P, _T]) -> type[Callable[_P, _T]]:
```

Comment:
Woah, this is wild. So this is actually invoked for just: `Partial[MyClass]`?

Reply:
YEah!